pygraphviz/agraph.py

```diff
@@ -230,6 +230,9 @@ def __repr__(self):
             return f"<AGraph {self.handle}>"
         return f"<AGraph {name} {self.handle}>"
 
+    def _repr_svg_(self):
+        return self.draw(format='svg').decode(self.encoding)
+
```

Comment:
As it stands, this will give an `AttributeError` when users try to display an `AGraph` instance in a jupyter notebook cell when `layout` is called, which is a regression that has the potential to break a lot of existing notebooks.

I can think of a couple options: one would be to catch the `AttributeError` and return the original `repr` when the graph hasn't yet had its layout computed, e.g.:

```python
    def _repr_svg_(self):
        try:
            return self.draw(format='svg').decode(self.encoding)
        except AttributeError:
            return repr(self)
```

Another possible solution is to compute the layout inside `_repr_svg` in the case where it hasn't already been done. The biggest downside there is that the `layout` method might be computationally expensive depending on the method used and the size of the graph, which means executing a cell for a seemingly simple repr might result in unexpected behavior for users.

I think I'm partial to the first solution since it's the least invasive and avoids any major assumptions about what the user intended, but I'm curious to hear other opinions!

Reply:
Another option would be to implement [`_repr_mimebundle_()`](https://ipython.readthedocs.io/en/stable/config/integrating.html#MyObject._repr_mimebundle_), and switch representations based on whether or not layout has been called yet; include `image/svg+xml` if layout has been called, `text/plain` when not?

Reply:
@rossbar Thanks for catching this.  @mjpieters I like that idea.  I also like the two ideas of @rossbar, I just hadn't got a chance to parse out which I preferred yet.

Now that I see the `_repr_mimebundle_()` suggestion, I think I like that the best, personally.  It seems to me this would allow the user to get the desired behavior -- whether text repr or layout + SVG repr -- in the most explicit/intuitive way, while not breaking the existing use cases.

I'd be happy to work at implementing that (or either of the others, whatever is decided to be the best).  To determine if `layout()` has been called or not, would it be sufficient to check `self.has_layout`?